.changeset/good-camels-pull.md

```diff
@@ -0,0 +1,56 @@
+---
+'@astrojs/markdoc': minor
+'@astrojs/mdx': major
+'@astrojs/markdown-remark': major
+'astro': major
+---
+
+Updates Markdown heading ID generation
+
+In Astro 5.x, an additional default processing step to Markdown stripped trailing hyphens from the end of IDs for section headings ending in special characters. This provided a cleaner `id` value, but could lead to incompatibilities rendering your Markdown across platforms.
+
+In Astro 5.5, the `experimental.headingIdCompat` flag was introduced to allow you to make the IDs generated by Astro for Markdown headings compatible with common platforms like GitHub and npm, using the popular [`github-slugger`](https://github.com/Flet/github-slugger) package.
+
+Astro 6.0 removes this experimental flag and makes this the new default behavior in Astro: trailing hyphens from the end of IDs for headings ending in special characters are no longer removed.
+
+#### What should I do?
+
+If you have manual links to headings, you may need to update the anchor link value with a new trailing hyphen. For example, the following Markdown heading:
+
+```md
+## `<Picture />`
+```
+
+will now generate the following HTML with a trailing hyphen in the heading `id`:
+
+```html
+<h2 id="picture-"><code>&lt;Picture /&gt;</h2>
+```
+
+and must now be linked to as:
+
+```markdown
+See [the Picture component](/en/guides/images/#picture-) for more details.
+```
+
+If you were previously using the experimental feature to enforce trailing hyphens, you must [remove this experimental flag from your configuration](#experimental-flags) as it no longer exists.
```

Comment:
```suggestion
If you were previously using the experimental feature to enforce trailing hyphens, you must remove this experimental flag from your configuration as it no longer exists.
```
Sadly, we can't rely on the docs upgrade guide trick here, and I guess probably have to show removing the flag?